### PR TITLE
Add home page

### PR DESCRIPTION
**Pull request by BaptisteLegat**

### CODE REVIEW

**Review by Baptiste-Ferrand — APPROVED** [review]

File: app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    body: "I've tried many trip planners, but this one is by far the best. The AI customization feature really understands my preferences and suggests perfect trips. The dashboard is also very informative and helps me track my travel goals.",
+    img: "https://avatar.vercel.sh/sarahthompson",
+  },
+  {
+    name: 'David Wilson',
+    username: '@davidwilson',
+    body: "I am amazed at how intuitive and user-friendly this trip planner is. It has made trip planning a fun and enjoyable experience. The dashboard with statistics is excellent motivation to keep exploring and traveling.",
+    img: "https://avatar.vercel.sh/davidwilson",
+  },
+];
```
Comment (Dercraker) [inline app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx:36]: Dont store fake data or data in component

File: app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      <Marquee reverse pauseOnHover className="[--duration:20s]">
+        {secondRow.map((review) => (
+          <ReviewCard key={review.username} {...review} />
+        ))}
+      </Marquee>
+      <div className="pointer-events-none absolute inset-y-0 left-0 w-1/3"></div>
+      <div className="pointer-events-none absolute inset-y-0 right-0 w-1/"></div>
+    </div>
+  );
+};
```
Comment (Dercraker) [inline app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx:94]: One component by file

File: app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  {
+    name: 'David Wilson',
+    username: '@davidwilson',
+    body: "I am amazed at how intuitive and user-friendly this trip planner is. It has made trip planning a fun and enjoyable experience. The dashboard with statistics is excellent motivation to keep exploring and traveling.",
+    img: "https://avatar.vercel.sh/davidwilson",
+  },
+];
+
+const firstRow = reviews.slice(0, reviews.length / 2);
+const secondRow = reviews.slice(reviews.length / 2);
```
Comment (Dercraker) [inline app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx:39]: write this inside of the component

File: app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx:1]: Change name ClientReviews must be clientReviews

File: app/(landing-layout)/_component/FeaturesSection.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  },
+  {
+    Icon: IconDownload as React.ElementType,
+    name: "Download your itinerary",
+    description: "Download your itinerary for offline use",
+    href: "/",
+    cta: "Learn more",
+    className: "lg:row-start-5 lg:row-end-6 lg:col-start-3 lg:col-end-4",
+  },
+];
```
Comment (Dercraker) [inline app/(landing-layout)/_component/FeaturesSection.tsx:108]: don't store data or fake data in component.

Either you create a seed with prisma, to faker the data in the db. Or you create a ts file in src/features/Home/features.ts which returns the data.

File: app/(landing-layout)/_component/FeaturesSection.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+          </Text>
+        </Stack>
+        <BentoGrid className="lg:grid-rows-3 mb-3">
+          {features.map((feature) => (
+            <BentoCard key={feature.name} {...feature} />
+          ))}
+        </BentoGrid>
+        <Divider size={2} color="teal" mt="xl" />
+      </Container>
+    </>
```
Comment (Dercraker) [inline app/(landing-layout)/_component/FeaturesSection.tsx:129]: Here the artifact are useless

```suggestion
      <Container size="xl" mt="sm">
        <Stack w="100%" maw={400}>
          <Title order={1} c="teal" fw={900}>Features</Title>
        </Stack>
        <Stack w="100%" maw={600}>
          <Text size="lg" c="teal" my="sm">
            Here is the list of features included in our application and which you can discover.
          </Text>
        </Stack>
        <BentoGrid className="lg:grid-rows-3 mb-3">
          {features.map((feature) => (
            <BentoCard key={feature.name} {...feature} />
          ))}
        </BentoGrid>
        <Divider size={2} color="teal" mt="xl" />
      </Container>
```

File: app/(landing-layout)/_component/FeaturesSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/FeaturesSection.tsx:1]: The file name must be respect the camelCase convetion -> featuresSection.tsx

File: app/(landing-layout)/_component/GlobeSection.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  Divider,
+  Grid,
+  GridCol,
+  Flex,
+} from '@mantine/core';
+import Globe from '@/components/magicui/globe';
+
+export const GlobeSection = () => {
+  return (
+    <>
```
Comment (Dercraker) [inline app/(landing-layout)/_component/GlobeSection.tsx:14]: useless artifact

File: app/(landing-layout)/_component/GlobeSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/GlobeSection.tsx:1]: camelCase name for files

File: app/(landing-layout)/_component/GlobeSection.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+              </Text>
+            </Flex>
+            <Title order={2} fw={700} mt={{ xs: 20, md: 'xl' }}>
+              Confirm and the map updates!
+            </Title>
+          </GridCol>
+        </Grid>
+        <Divider size={2} color="teal" mt="xl" />
+      </Container>
+    </>
```
Comment (Dercraker) [inline app/(landing-layout)/_component/GlobeSection.tsx:46]: useless artifact

File: app/(landing-layout)/_component/GlobeSection.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+} from '@mantine/core';
+import Globe from '@/components/magicui/globe';
+
+export const GlobeSection = () => {
+  return (
+    <>
+      <Container size="xl" py={{ xs: 20, md: 50 }}>
+        <Grid gutter="xl">
+          <GridCol span={{ xs: 12, md: 6 }} mt={{ base: 20, md: 0 }}>
+            <div className="relative flex h-full w-full max-w-[32rem] items-center justify-center overflow-hidden rounded-lg px-40 pb-40 pt-8 md:pb-60">
```
Comment (Dercraker) [inline app/(landing-layout)/_component/GlobeSection.tsx:18]: for flex flex-row you can use [Group](https://mantine.dev/core/group/) and for flex flex-col you can use [Stack](https://mantine.dev/core/stack/)

File: app/(landing-layout)/_component/GlobeSection.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+              </Text>
+              <Text>
+                <strong className="text-xl">Dates</strong><br />
+                Select the dates that suit you
+              </Text>
+              <Text>
+                <strong className="text-xl">Details</strong><br />
+                Add the details of your step
+              </Text>
+            </Flex>
```
Comment (Dercraker) [inline app/(landing-layout)/_component/GlobeSection.tsx:38]: for the flex row you can use group

File: app/(landing-layout)/_component/GlobeSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/GlobeSection.tsx:1]: camelCase for the file name

File: app/(landing-layout)/_component/Hero/HeroHeader.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import { Button, Container, Text, Title, Overlay } from '@mantine/core';
+import { useRouter } from 'next/navigation';
+import './HeroImageBackground.scss';
```
Comment (Dercraker) [inline app/(landing-layout)/_component/Hero/HeroHeader.tsx:5]: use mantine or tailwind before scss or css file

File: app/(landing-layout)/_component/Hero/HeroHeader.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+import { Video } from '@/components/video/Video';
+
+export const HeroImageBackground: React.FC = () => {
+  const router = useRouter();
+
+  const handleButtonClick = () => {
+    router.push('/trips');
```
Comment (Dercraker) [inline app/(landing-layout)/_component/Hero/HeroHeader.tsx:12]: use the constant LINKS to provide the href

File: app/(landing-layout)/_component/Hero/HeroHeader.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/Hero/HeroHeader.tsx:1]: camelCase

File: app/(landing-layout)/_component/Marquee/Marquee.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/Marquee/Marquee.tsx:1]: remove the empty file

File: app/(landing-layout)/_component/Hero/HeroImageBackground.scss
Comment (Dercraker) [inline app/(landing-layout)/_component/Hero/HeroImageBackground.scss:1]: camelCase

File: app/(landing-layout)/_component/StepSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/StepSection.tsx:1]: camelCase

File: app/(landing-layout)/_component/TimelineAnimation/TimelineAnimation.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/TimelineAnimation/TimelineAnimation.tsx:1]: camelCase

File: app/(landing-layout)/_component/TestimonialsSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/TestimonialsSection.tsx:1]: camelCase

File: playwright-report/index.html
Comment (Dercraker) [inline playwright-report/index.html:1]: gitignore ?

File: src/components/video/Video.tsx
Comment (Dercraker) [inline src/components/video/Video.tsx:1]: camelCase

File: src/components/video/Video.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+import React from "react";
+import ReactPlayer from "react-player";
+import { Box } from '@mantine/core';
+
+type VideoProps = {
+    source: string;
+    children?: ReactNode;
+}
+
+export const Video: React.FC<VideoProps> = ({ source, children }) => {
```
Comment (Dercraker) [inline src/components/video/Video.tsx:11]: why React.FC<VideoProps> ?

File: src/hook/useScrollTrigger.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+          end: 'bottom top',
+          scrub: 1,
+          toggleActions: 'play none none reverse',
+        },
+      });
+    });
+  }, []);
+};
+
+export default useScrollTrigger;
```
Comment (Dercraker) [inline src/hook/useScrollTrigger.ts:30]: prefer export const .... than export default for hooks

File: src/lib/utils.ts [hunk 14]
```diff
@@ -1,6 +1,6 @@
...
-import { twMerge } from 'tailwind-merge';
+import { type ClassValue, clsx } from "clsx"
+import { twMerge } from "tailwind-merge"
 
-export const cn = (...inputs: ClassValue[]) => {
-  return twMerge(clsx(inputs));
-};
+export function cn(...inputs: ClassValue[]) {
+  return twMerge(clsx(inputs))
+}
```
Comment (Dercraker) [inline src/lib/utils.ts:6]: we use arrow function like const funcName = (somes params) => {}

Cancel your edit

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx (hunk 1 above)
Comment (BaptisteLegat) [reply app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx:36]: Waiting [feat feedback](https://github.com/From-A2B/WebApp/pull/29) merge

File: app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/ClientReviews/ClientReviews.tsx:1]: camelCase

File: app/(landing-layout)/_component/ClientReviews/reviewCard.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/ClientReviews/reviewCard.tsx:1]: camelCase

File: app/(landing-layout)/_component/FeaturesSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/FeaturesSection.tsx:1]: camelCase

File: app/(landing-layout)/_component/GlobeSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/GlobeSection.tsx:1]: camelCase

File: app/(landing-layout)/_component/Hero/HeroHeader.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/Hero/HeroHeader.tsx:1]: camelCase

File: app/(landing-layout)/_component/StepSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/StepSection.tsx:1]: camelCase

File: app/(landing-layout)/_component/TestimonialsSection.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/TestimonialsSection.tsx:1]: camelCase

File: app/(landing-layout)/_component/TestimonialsSection.tsx [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+  Container,
+  Stack,
+  Text,
+  Title,
+} from '@mantine/core';
+import { ClientReviews } from './clientReviews/clientReviews';
+
+export const TestimonialsSection = () => {
+  return (
+    <>
```
Comment (Dercraker) [inline app/(landing-layout)/_component/TestimonialsSection.tsx:11]: useless artifact

File: app/(landing-layout)/_component/TimelineAnimation/TimelineAnimation.tsx
Comment (Dercraker) [inline app/(landing-layout)/_component/TimelineAnimation/TimelineAnimation.tsx:1]: camelCase

File: src/components/video/Video.tsx
Comment (Dercraker) [inline src/components/video/Video.tsx:1]: camelCase

File: app/(base-layout)/feedback/_component/feedbackCarousel.tsx [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+  if (error || !feedbacks) {
+    return (
+      <Center>
+        <Loader size="xl" mt="xl" />
+      </Center>
+    );
+  }
+
+  return (
+    <>
```
Comment (Dercraker) [inline app/(base-layout)/feedback/_component/feedbackCarousel.tsx:51]: useless artifact

File: app/(base-layout)/support/_component/contact.form.tsx [hunk 17]
```diff
@@ -1,7 +1,7 @@
 'use client';
 
 import { CheckIcon } from '@/components/icons/check.icon';
-import { TrashIcon } from '@/components/icons/trash.icon';
+import { TrashIcon } from '~/public/images/trash.icon';
```
Comment (Dercraker) [inline app/(base-layout)/support/_component/contact.form.tsx:4]:
```suggestion
import { TrashIcon } from '@/components/icons/trash.icon';
```

File: app/(landing-layout)/_component/header/header.module.css [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+
+  backdrop-filter: blur(12px);
+
+  @mixin hover {
+    background-color: light-dark(
+      var(--mantine-color-gray-0),
+      var(--mantine-color-dark-6)
+    );
+  }
+}
```
Comment (Dercraker) [inline app/(landing-layout)/_component/header/header.module.css:15]: tu peut pas utiliser mantine ou tailwind pour sa ?

File: src/components/contact/support/ContactSupportDialog.tsx [hunk 19]
```diff
@@ -1,6 +1,6 @@
 'use client';
 
-import { TrashIcon } from '@/components/icons/trash.icon';
+import { TrashIcon } from '~/public/images/trash.icon';
```
Comment (Dercraker) [inline src/components/contact/support/ContactSupportDialog.tsx:3]:
```suggestion
import { TrashIcon } from '@/components/icons/trash.icon';
```

File: src/components/trip/createTripModal.tsx [hunk 20]
```diff
@@ -10,16 +10,16 @@ import {
 } from '@mantine/core';
 import { useDisclosure } from '@mantine/hooks';
 
-import { CheckIcon } from '@/components/Icons/checkIcon';
-import { TrashIcon } from '@/components/Icons/trashIcon';
+import { CheckIcon } from '~/public/images/checkIcon';
+import { TrashIcon } from '~/public/images/trashIcon';
```
Comment (Dercraker) [inline src/components/trip/createTripModal.tsx:14]:
```suggestion
import { CheckIcon } from '@/components/Icons/checkIcon';
import { TrashIcon } from '@/components/Icons/trashIcon';
```

File: src/components/trip/createTripModal.tsx [hunk 21]
```diff
@@ -10,16 +10,16 @@ import {
...
 import { CreateTripAction } from '@/features/trips/create/createTrip.action';
 import { CreateTripSchema } from '@/features/trips/create/createTrip.schema';
 import useNotify from '@/hook/useNotify';
 import { DatePickerInput } from '@mantine/dates';
 import { useForm, zodResolver } from '@mantine/form';
 import { useMutation } from '@tanstack/react-query';
 import moment from 'moment';
-import { CrossCircleIcon } from '../Icons/crossCircleIcon';
+import { CrossCircleIcon } from '../icons/crossCircle.icon';
```
Comment (Dercraker) [inline src/components/trip/createTripModal.tsx:22]:
```suggestion
import { CrossCircleIcon } from '@/components/Icons/crossCircle.icon';
```

File: src/utils/NavigationLinks.tsx [hunk 22]
```diff
@@ -1,7 +1,7 @@
 import { EmailIcon } from '@/components/icons/email.icon';
 import { MoneyIcon } from '@/components/icons/money.icon';
-import { UserIcon } from '@/components/icons/user.icon';
-import { WarnIcon } from '@/components/icons/warn.icon';
+import { UserIcon } from '~/public/images/user.icon';
+import { WarnIcon } from '~/public/images/warn.icon';
```
Comment (Dercraker) [inline src/utils/NavigationLinks.tsx:4]:
```suggestion
import { UserIcon } from '@/components/icons/user.icon';
import { WarnIcon } from '@/components/icons/warn.icon';
```

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: app/(landing-layout)/_component/header/header.module.css (hunk 18 above)
Comment (BaptisteLegat) [reply app/(landing-layout)/_component/header/header.module.css:15]: non

File: app/(base-layout)/faq/page.tsx [hunk 23]
```diff
@@ -1,3 +1,5 @@
+'use client';
```
Comment (Dercraker) [inline app/(base-layout)/faq/page.tsx]: useless

**Review by Dercraker — APPROVED** [review]

**Pull request merged**